script.js

```diff
@@ -1,54 +1,71 @@
+(function(global) {
 
-function domobj(){
-  var self        =this;
-  self.products   = [];
+  var requestData = $.getJSON('data.json').done(function(products) {
+    var requestProductTemplate = $.get('product-template.html').done(function(template) {
 
-  self.getproducts = function(url){
-    $.getJSON(url, function(response){
-        for(i=0; i<response.sales.length ; i++){
-          self.products.push( new productobj(response.sales[i], i)  );
+      var data = products;
+      var template = template;
+
+      function TOMApp() {
+        this.products = [];
+      }
+
+      TOMApp.prototype.getproducts = function() {
```

Comment:
Pseudo-classical pattern prevents functions being created many times when instantiating the object.